lib/puppet/forge/repository.rb

```diff
@@ -109,7 +110,7 @@ def read_response(request, io = nil)
         end
 
         io.write(response.body) if io.respond_to? :write
-        response
+        return response
```

Comment:
And of course if I change back to `return response`, above, the second breakpoint never gets hit.